### ISSUE TITLE
Conditional generation

### ISSUE THREAD

**Issue by abdalazizrashid** [lucidrains/denoising-diffusion-pytorch#7]
Is it possible to use this for conditional generation?

**Comment by alshenoudy** [comment]
@abdalazizrashid This was transcribed from the original TensorFlow implementation to Pytorch and it does not support conditional generation. 

Take a look at the this [paper](https://openreview.net/pdf?id=FuLL40HLCRn), they introduced a condition encoding network which you can implement based on the code here by a few modifications to some functions and the loss to accomodate for this new condition.

Hope it helps.

**Comment by FernandoCamaro** [comment]
I think that adding the conditional input is quite easy.
You could have a look to section 5 of this paper:
https://arxiv.org/pdf/2102.09672.pdf

**Comment by sgbaird** [comment]
If anyone tries this out, would be great if you could link to some code or provide a MWE. Or if anyone is aware of other implementations

**Comment by malekinho8** [comment]
Has anyone been able to apply this code for tasks like inpainting or super-resolution? They are both forms of conditional generation, and it's not clear to me how one should go about doing it.

**Comment by sgbaird** [comment]
See also https://github.com/lucidrains/denoising-diffusion-pytorch/issues/21#issuecomment-1143989193

**Comment by ifding** [comment]
Class-conditional generation can be found in <https://github.com/openai/improved-diffusion>.

**Comment by B-guppy** [comment]
Just wondering if anyone had an update on this. I was going through the SR3 and Palette paper and I think I am going to try the approach of taking the source image and concatenating it with $x_{t}$ along the channel dimension and see whether that works.
